bson/value_reader.go

```diff
@@ -750,78 +737,44 @@ func (vr *valueReader) readBytes(length int32) ([]byte, error) {
 		return nil, fmt.Errorf("invalid length: %d", length)
 	}
 
-	n, err := vr.prepload(length)
-	if n < length || err != nil {
-		if err == nil {
-			err = io.EOF
-		}
+	buf := make([]byte, length)
```

Comment:
Optional: The `length` is frequently static, which allows using static-size arrays or slices. To take advantage of that, consider changing the `readBytes` API to accept a `[]byte` instead of a length.

E.g.
```go
func (vr *valueReader) read(buf []byte) (int, error) { ... }
```

Then an example call from `readi32` would look like:
```go
func (vr *valueReader) readi32() (int32, error) {
	var buf [4]byte
	_, err := vr.read(buf[:])
	// ...
```